### PR TITLE
Add 100+ fonts from jave.de Figlet Fonts collection v4.0

### PR DESCRIPTION
**Pull request by MinchinWeb**
these fonts are taken from http://www.jave.de/figlet/fonts.html

### CODE REVIEW

**Comment by pwaller** [comment]
1. What is the license on these font files?
 - Can you find out from the Author if we need to include a special license.txt file for them?
2. You put modifications which were not adding fonts in (changes to .gitignore and travis)
 - Can you please submit those as a separate PR?

Thanks!

P.S. I'm fairly busy so if I don't respond for a couple of weeks please send pings!

**Comment by pwaller** [comment]
Ping!

**Comment by MinchinWeb** [comment]
```
Pinging MinchinWeb with 32 bytes of data:
Reply from MinchinWeb: time=846661801ms
```

Jokes aside, figuring out the licensing is rather annoying. The original download is [here](http://www.jave.de/figlet/fonts.html): the zip file is almost 8 years old, has multiple authors, appears to be a complication from several sources going back 20+ years, and doesn't include a `license.txt` file of it's own.

## Figlet 2.0 ([New BSD License](http://www.figlet.org/figlet_history.html)?)
- arrows.flf
- greek.flf
- shimrod.flf
- tiles.flf
- tubular.flf

## Figlet 2.1 ([New BSD License](http://www.figlet.org/figlet_history.html)?)
- lower.flc
- null.flc
- rot13.flc
- upper.flc

## Figlet 2.2 ([New BSD License](http://www.figlet.org/figlet_history.html)?)
- defleppard.flf

## Figlet 2.2.2 ([New BSD License](http://www.figlet.org/figlet_history.html)?)
- wavy.flf

## Various Authors, permissive licensing in raw file
> Permission is hereby given to modify this font, as long as the
> modifier's name is placed on a comment line.

- 3d_diagonal.flf
- B1FF.flf
- ICL-1900.flf
- ascii_new_roman.flf
- bear.flf
- bigfig.flf
- blocks.flf
- braced.flf
- broadway_kb.flf
- cards.flf
- chiseled.flf
- cola.flf
- crazy.flf
- dancingfont.flf
- dietcola.flf
- doubleshorts.flf (based on a Usenet message from 1994)
- filter.flf
- fire-font-k.flf
- fire-font-s.flf
- flipped.flf
- flowerpower.flf
- funface.flf
- funfaces.flf
- ghost.flf
- ghouslish.flf
- glenyn.flf (commercial use may be restricted)
- heart_left.flf
- heart_right.flf
- hieroglyphs.flf
- horizontalleft.flf
- horizontalright.flf
- impossible.flf
- knob.flf
- konto.flf
- kontoslant.flf
- lildevil.flf
- lineblocks.flf
- merlin1.flf
- merlin2.flf
- modular.flf
- morse2.flf (slightly different license statement)
- muzzle.flf
- puzzle.flf
- rammstein.flf
- rotated.flf
- s-relief.flf
- smallcaps.flf
- soft.flf
- starstrips.flf
- swampland.flf
- sweet.flf
- test1.flf
- train.flf
- twisted.flf
- varsity.flf
- wetletter.flf
- wow.flf

## No Contact Leads
### Author Unknown
- 1row.flf

### Author 'known', Dead webpage
```
     .-~~-.
    (_^..^_)
Lester||||AMC - Anthony Cucchiara
```

- amc3line.flf
- amc3liv1.flf
- amcaaa01.flf
- amcneko.flf
- amcrazo2.flf
- amcrazor.flf
- amcslash.flf
- amcslder.flf
- amcthin.flf
- amctubes.flf
- amcun1.flf

### No Email Given
> by Mike Rosulek. 1/12/95

- bolger.flf (called fatty.flf in file header)

> Dennis Monk  7/94

- bright.flf

> Christian 'CeeJay' Jensen

- cygnet.flf
- swan.flf

> Richard Sabey 8.2003

- georgi16.flf

## Email given
> by Philip Menke (philippe@dds.nl)
> April 1995

- 4max.flf

> Marcin `stforek` Glinski

- stforek.flf

> DANC4 by Richard Sabey <cryptic_fan@hotmail.com> 9.2003

- DANC4.flf

> Alligator2 by Daniel Wiz. AKA Merlin Greywolf <merlin@brahms.udel.edu>
> 27th July, 1994
>    
> Changed 2004-02-18: Bugfix in '\' character by Markus Gebhard

- alligator3.flf

```
                    __ :|                |                  |
                   |__|:|   .---.        | Lennert Stock    |
                   :::::|  /  oo|        | UvA/FWI          |
                 ::'____|=' ( _\|        | stock@fwi.uva.nl |
                :::|_ __| `-----' -======:__________________|
                     |__|
                        |
```
- alpha.flf

> Christian Garbs <mitch@cgarbs.de>

- benjamin.flf

> Rebel by Valerie Mates (popcorn@cyberspace.org), based on a font by Ron Bliss
> (who sometimes goes by the name "rebel" because his initials are REB). [...]
> May 27, 1994

- dosrebel.flf

> Characters by Henry Segerman henryseg@email.com,
> Converted to FIGlet font by Markus Gebhard markus@jave.de

- henry3d.flf

> http://studenten.freepage.de/meph/ascii/eng/eng.htm

- jacky.flf

> composed by Vinney Thai <ssfiit@eris.cc.umb.edu>
> date: Nov 21, 1994

- keyboard.flf

> vampyr@acs.bu.edu

- nancyj-improved.flf

> by Normand Veilleux <nveilleu@emr.ca> Oct 7, 1994
> figletized by Peter Samuelson <psamuels@hmc.edu> Nov 29, 1994

- nscript.flf

> Ryan Youck (youck@cs.uregina.ca) [...] March 7, 1994

- oldbanner.flf

> Victor Parada (vparada@inf.utfsm.cl) 94/09/23

- peaksslant.flf

> Purdie@zeus.usq.edu.au (Red Phoenix)

- red_phoenix.flf

> Matt E. Thurston(thurstme@cda.mrs.umn.edu)

- reverse.flf

> Derived from a copyrighted program by Jan Wolter (janc@crim.eecs.umich.edu)
> Figlet-ized by Wendell Hicken (whicken@parasoft.com)

- santaclara.flf

> figletized by Vinney Thai <ssfiit@eris.cc.umb.edu>
> poison font composed David Issel <dissel@nunic.nu.edu>
> date: Nov 18, 1994

- smpoison.flf

> nathan bloomfield (xzovik@gmail.com) (and Markus Gebhard?)

- spliff.flf

> Marco Bodrato <bodrato@genio.sns.it>
> Data: 12 Novembre 1994

- stampate.flf

> "Sub-Zero" <bodom@papaya.ucs.indiana.edu>

- sub-zero.flf

**Comment by MinchinWeb** [comment]
The commits have been force pushed here, now removing the Travis-CI addition of Python 3.5 (now in Pull Request #48) and changes to `.gitignore`.

**Comment by pwaller** [comment]
Thanks for following up looks like you put in a fair effort there!. Please bear with me while I go on holiday. Ping me if I don't respond within two weeks. ☺

On 30 January 2016 20:38:44 GMT+00:00, MinchinWeb <notifications@github.com> wrote:
>The commits have been force pushed here, now removing the Travis-CI
>addition of Python 3.5 (now in Pull Request #48) and changes to
>`.gitignore`.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/pwaller/pyfiglet/pull/46#issuecomment-177295572

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by MinchinWeb** [comment]
Ping! Any progress on your end?

**Comment by MinchinWeb** [comment]
ping? Hope your holiday was amazing!

File: pyfiglet/test.py [hunk 1]
```diff
@@ -48,7 +51,13 @@ def outputUsingFigletorToilet(self, text, font, fontpath):
...
 
         p = Popen(cmd, bufsize=4096, stdout=PIPE)
-        outputFiglet = p.communicate()[0].decode('utf8')
+        try:
+            outputFiglet = p.communicate()[0].decode('utf8')
+        except UnicodeDecodeError as e:
+            print("Unicode Error handling font {}".format(font))
+            outputFiglet = ''
+        except e:
+            raise e
```
Comment (pwaller) [inline pyfiglet/test.py]: This `except e: raise e` looks superfluous?

File: pyfiglet/test.py [hunk 2]
```diff
@@ -33,10 +33,13 @@ def __init__(self, opts):
         self.fail = 0
         self.failed = []
         self.oked = []
-        self.skip = ['runic','pyramid','eftifont']  # known bug..
+        # known bugs..
+        self.skip = ['runic', 'pyramid', 'eftifont', 'DANC4', 'dietcola']
```
Comment (pwaller) [inline pyfiglet/test.py]: This line has extra whitespace at the end, please remove it.

**Comment by pwaller** [comment]
@stefanor would these extra fonts cause any problems for Debian?

LGTM other than minor nits. Will merge and make a release fairly rapidly, especially if @stefanor says it's OK.

File: pyfiglet/test.py (hunk 1 above)
Comment (MinchinWeb) [inline pyfiglet/test.py]: Probably. It if was anything other than a UnicodeDecodeError, I didn't want it to go by un-noticed.

File: pyfiglet/test.py (hunk 2 above)
Comment (MinchinWeb) [inline pyfiglet/test.py]: Fixed in 7e9c832

File: pyfiglet/test.py (hunk 1 above)
Comment (MinchinWeb) [inline pyfiglet/test.py]: Removed in 7e9c832

**Comment by MinchinWeb** [comment]
Sounds great. I'll leave it in your capable hands!

**Comment by pwaller** [comment]
v0.7.5 released with this in it. Thanks for your patience.

**Comment by pwaller** [comment]
(Also, please test and let me know if you spot anything wrong :)

**Comment by MinchinWeb** [comment]
I've downloaded it off PyPI and done some quick testing and everything seems to be working as expected!

**Pull request merged**